akka-cluster/src/main/scala/akka/cluster/ClusterRemoteWatcher.scala

```diff
@@ -102,10 +102,8 @@ private[cluster] class ClusterRemoteWatcher(
    * by super RemoteWatcher.
    */
   def takeOverResponsibility(address: Address): Unit = {
-    watching foreach {
-      case (watchee, watcher) ⇒ if (watchee.path.address == address)
-        unwatchRemote(watchee, watcher)
-    }
+    watching.keys.filter(_.path.address == address).foreach(clearAllWatches)
```

Comment:
also, I'm not sure, but I think it's more efficient to use `foreach { case (key, value)` than `keys`, because `keys` must create some kind of intermediate collection